broker/query_plan_non_agg.go

```diff
@@ -148,12 +157,46 @@ func (nqp *NonAggQueryPlan) Execute(ctx context.Context) (err error) {
 			return
 		}
 		// write rows
-		nqp.w.Write(res.data)
-		if i != len(nqp.nodes)-1 {
+		if nqp.limit < 0 {
+			// when no limit, flush data directly
+			nqp.w.Write(res.data)
+		} else {
+			res.data = append([]byte("["), res.data[:]...)
```

Comment:
add some more test cases to simulate multiple nodes return and merge?

Reply:
it's in the query_plan_non_agg_test.go. simulated 3 nodes.